test/recipes/80-test_ca.t

```diff
@@ -80,3 +232,53 @@ sub yes {
     return 0;
 }
 
+# Get the value of the lastUpdate or nextUpdate field from a CRL
+sub crl_field {
+    my ($crl_path, $field_name) = @_;
+
+    my @out = run(
+        app(['openssl',
+             'crl',
+             '-in', $crl_path,
+             '-noout',
+             '-' . lc($field_name),
+        ]),
+        capture => 1,
+        statusvar => \my $exit,
+    );
+    ok($exit, "CRL $field_name field retrieved");
+    diag("CRL $field_name: $out[0]");
+
+    $out[0] =~ s/^\Q$field_name\E=//;
+    $out[0] =~ s/\n?//;
+    my $time = human_time($out[0]);
+
+    return $time;
+}
+
+# Converts human-readable ASN1_TIME_print() output to Unix time
+sub human_time {
+    my ($human) = @_;
+
+    my ($mo, $d, $h, $m, $s, $y) = $human =~ /^([A-Za-z]{3})\s+(\d+) (\d{2}):(\d{2}):(\d{2}) (\d{4})/;
+
+    my %months = (
+        Jan => 0, Feb => 1, Mar => 2, Apr => 3, May => 4,  Jun => 5,
+        Jul => 6, Aug => 7, Sep => 8, Oct => 9, Nov => 10, Dec => 11,
+    );
+
+    return timegm($s, $m, $h, $d, $months{$mo}, $y);
+}
+
+# Converts an RFC 5280 timestamp to Unix time
+sub rfc5280_time {
+    my ($asn1) = @_;
+
+    my ($y, $mo, $d, $h, $m, $s) = $asn1 =~ /^(\d{2,4})(\d{2})(\d{2})(\d{2})(\d{2})(\d{2})Z$/;
+
+    return timegm($s, $m, $h, $d, $mo - 1, $y);
+}
+
+sub demoCA_file {
+    return srctop_file('test-runs', 'test_ca', 'demoCA', @_);
```

Comment:
I cant see anywhere else that accesses test-runs like this for example

Reply:
Yeah, I think I was trying to write files to a path that doesn't exist in every CI environment by doing that. The latest commit should fix that by not caring too much about where temporary CSRs/certificates/CRLs are written (no other test in this file appears to, so hopefully that's not a problem).